src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -303,6 +299,13 @@ public override async ValueTask<FormulaValue> Visit(CallNode node, EvalVisitorCo
 
             CheckCancel();
             return result;
+        }
+
+        private bool TryGetServiceFunctionPtr(TexlFunction func, out AsyncFunctionPtr ptr)
+        {
+            PowerFxFunctionPtr pfxFunctionPtr = (PowerFxFunctionPtr)_services.GetService(typeof(PowerFxFunctionPtr<>).MakeGenericType(func.GetType()));

```

Comment:
>MakeGenericType [](http://example.com/codeflow?start=118&length=15)

We shouldn't need to call MakeGenericType to do a lookup on a hotpath. Let's chat about alternatives.

Reply:
Ok.